### PR TITLE
SB-833: Nuget `push` fixed

### PR DESCRIPTION
**Pull request by sbespalov**
+ [SB-835] `Authenticator` beans initialization fixed

### CODE REVIEW

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java [hunk 1]
```diff
@@ -70,26 +70,17 @@ public void scanAndReloadRegistry()
     private List<Authenticator> getAuthenticators(ClassLoader currentClassLoader,
                                                   ApplicationContext applicationContext)
     {
-        final List<String> authenticatorsClasses = applicationContext.getBean("authenticators", List.class);
+        final List<Object> authenticatorsClasses = applicationContext.getBean("authenticators", List.class);
         final List<Authenticator> authenticators = new ArrayList<>();
-        for (final String auth : authenticatorsClasses)
+        for (final Object authenticator : authenticatorsClasses)
```
Comment (sbespalov) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java]: @fuss86 pleace note this changes
see [SB-835](https://youtrack.carlspring.org/issue/SB-835) for details

**Comment by sbespalov** [comment]
@fuss86 
can you please tell where can I find sources for `strongbox-example-authentication-provider-1.0-SNAPSHOT.jar`?

Seems that I need to modify it too because fo Exception below:

```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.carlspring.strongbox.authentication.impl.example.EmptyAuthenticator#2651e783': Unsatisfied dependen
cy expressed through field 'authenticationProvider'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.security.authentication.A
uthenticationProvider' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.carlspring.strongbox.authentication.impl.example.EmptyAuthenticationProviderCompone
nt$EmptyAuthenticationProviderQ(), @javax.inject.Inject()}
```

**Comment by carlspring** [comment]
@sbespalov ,

You can find it [here](https://github.com/strongbox/strongbox-authentication-example).

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java (hunk 1 above)
Comment (fuss86) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java]: Hi @sbespalov, I see your improvements and they are good. However, I'm not sure if it's somehow related with [SB-835 ](https://youtrack.carlspring.org/issue/SB-835) ?

File: strongbox-user-management/src/test/resources/etc/conf/strongbox-authentication-providers.xml [hunk 2]
```diff
@@ -9,16 +9,14 @@
                            http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd
                            http://www.springframework.org/schema/security http://www.springframework.org/security/spring-security.xsd">
 
+    <context:annotation-config />
+
     <util:list id="authenticators" value-type="java.lang.String">
```
Comment (fuss86) [inline strongbox-user-management/src/test/resources/etc/conf/strongbox-authentication-providers.xml]: > value-type="java.lang.String">

is out-of-date if you accept your changes

File: strongbox-web-core/src/test/resources/etc/conf/strongbox-authentication-providers.xml [hunk 3]
```diff
@@ -7,10 +7,10 @@
                            http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd
                            http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd">
 
+    <context:annotation-config />
+
     <util:list id="authenticators" value-type="java.lang.String">
```
Comment (fuss86) [inline strongbox-web-core/src/test/resources/etc/conf/strongbox-authentication-providers.xml]: > value-type="java.lang.String">

as above

**Comment by fuss86** [comment]
@sbespalov, https://youtrack.carlspring.org/issue/SB-829 changes have been merged to master.

`strongbox-security-api` and `strongbox-user-management` have been relocated to the `strongbox-security` module under `masterbuild`. I guess above conflicts are related with these modules relocation so you just need to introduce your changes to the proper place now.

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java (hunk 1 above)
Comment (sbespalov) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java]: sorry, it's [SB-837](https://youtrack.carlspring.org/issue/SB-837)

**Comment by sbespalov** [comment]
@carlspring it can be merged

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetHierarchicalLayoutProvider.java [hunk 4]
```diff
@@ -173,6 +173,14 @@ protected ArtifactOutputStream decorateStream(RepositoryPath path,
         }
         
     }
+    
+    @Override
+    protected void addArtifactToIndex(RepositoryPath path)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetHierarchicalLayoutProvider.java]: This method should not exist in any of these implementations. We don't need empty methods that confuse things unnecessarily.

Also, the indexing is just for the Maven repositories. It's not for Nuget, for example. The `addToIndex()` is just for the Maven layout provider.

**Pull request merged**